go/store/prolly/sort/external_test.go

```diff
@@ -0,0 +1,504 @@
+			),
+			rows:      10_000,
+			batchSize: 10_000,
+			fileMax:   4,
+		},
+		{
+			name: "uint32",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.Uint32Enc, Nullable: false},
+			),
+			rows:      10_000,
+			batchSize: 1000,
+			fileMax:   4,
+		},
+		{
+			name: "uint32",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.Uint32Enc, Nullable: false},
+			),
+			rows:      20_000,
+			batchSize: 500,
+			fileMax:   16,
+		},
+		{
+			name: "int64",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.Int64Enc, Nullable: false},
+			),
+			rows:      7_777,
+			batchSize: 1000,
+			fileMax:   4,
+		},
+		{
+			name: "(string)",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.StringEnc, Nullable: false},
+			),
+			rows:      10_000,
+			batchSize: 100,
+			fileMax:   32,
+		},
+		{
+			name: "(string)",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.StringEnc, Nullable: false},
+			),
+			rows:      10_000,
+			batchSize: 483,
+			fileMax:   31,
+		},
+		{
+			name: "(string)",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.StringEnc, Nullable: false},
+			),
+			rows:      1,
+			batchSize: 100,
+			fileMax:   30,
+		},
+		{
+			name: "(string)",
+			td: val.NewTupleDescriptor(
+				val.Type{Enc: val.StringEnc, Nullable: false},
+			),
+			rows:      0,
+			batchSize: 100,
+			fileMax:   30,
+		},
+	}
+
+	tmpProv := newProv()
+	defer os.Remove(tmpProv.GetTempDir())
+	defer tmpProv.Clean()
+
+	ns := tree.NewTestNodeStore()
+
+	for _, tt := range tests {
+		t.Run(fmt.Sprintf("%s %d-rows %d-batch %d-files", tt.name, tt.rows, tt.batchSize, tt.fileMax), func(t *testing.T) {
+			keyCmp := func(l, r val.Tuple) bool {
+				return tt.td.Compare(l, r) <= 0
+			}
+
+			ctx := context.Background()
+			keys := testTuples(ns, tt.td, tt.rows)
+			s := NewTupleSorter(tt.batchSize, tt.fileMax, keyCmp, tmpProv)
+			expSize := 0
+			for _, k := range keys {
+				err := s.Insert(ctx, k)
+				require.NoError(t, err)
+				expSize += len(k)
+			}
+
+			iterable, err := s.Flush(ctx)
+			require.NoError(t, err)
+			var cnt, size int
+			iter, err := iterable.IterAll(ctx)
+			require.NoError(t, err)
+			var lastKey val.Tuple
+			for {
+				k, err := iter.Next(ctx)
+				if err != nil {
+					break
+				}
+				if lastKey != nil {
+					require.True(t, keyCmp(lastKey, k))
+				}
+				cnt++
+				size += len(k)
+				lastKey = k
+			}
+
+			require.Equal(t, tt.rows, cnt)
+			require.Equal(t, expSize, size)
+		})
+	}
+
+}
+
+func testTuples(ns tree.NodeStore, kd val.TupleDesc, cnt int) []val.Tuple {
+	keyBuilder := val.NewTupleBuilder(kd)
+
+	var keys []val.Tuple
+	for i := 0; i < cnt; i++ {
+		keys = append(keys, tree.RandomTuple(keyBuilder, ns))
+	}
+
+	return keys
+}
+
+func ensureSorted(t *testing.T, keys []val.Tuple, cmp func(val.Tuple, val.Tuple) bool) {
+	for i := 0; i < len(keys)-1; i += 2 {
+		require.True(t, cmp(keys[i], keys[i+1]))
+	}
+}
+
+func newProv() *tempfiles.TempFileProviderAt {
+	tmpDir := os.TempDir()
```

Comment:
I think prefer `t.TempDir()` on the `*testing.T` object for the test case, unless there is some reason we need to step out of that? `TempDir()` will automatically get cleaned up at the end of the test, but `defer` is fine too...